### ISSUE TITLE
Cannot read property 'htmlWebpackPluginAfterHtmlProcessing' of undefined

### ISSUE THREAD

**Issue by pmochine** [GoogleChromeLabs/critters#26]
Hey :)

Thanks for the plugin, however I would love to know why I have some troubles getting it to work.
I'm using Laravel Mix 3.0, but I get the error:

```
/Users/shadrix/Dropbox/Business/insta.food/Website_W_Laravel/instafood/node_modules/critters-webpack-plugin/dist/critters.js:140
        if (compilation.hooks.htmlWebpackPluginAfterHtmlProcessing) {
                              ^

TypeError: Cannot read property 'htmlWebpackPluginAfterHtmlProcessing' of undefined
    at Compiler.<anonymous> (/Users/shadrix/Dropbox/Business/insta.food/Website_W_Laravel/instafood/node_modules/critters-webpack-plugin/dist/critters.js:140:31)
    at Compiler.applyPlugins (/Users/shadrix/Dropbox/Business/insta.food/Website_W_Laravel/instafood/node_modules/tapable/lib/Tapable.js:61:14)

```

here are my settings

```
let mix = require('laravel-mix');

const webpack = require('webpack');

//https://github.com/ankurk91/laravel-mix-auto-extract
require('laravel-mix-auto-extract');

//https://github.com/GoogleChromeLabs/critters
const Critters = require('critters-webpack-plugin');

mix.js('resources/js/app.js', 'public/js')
	.js('resources/js/canvas.js', 'public/js')
	.js('resources/js/lib/moment.js', 'public/js') //to activate moment
	.sass('resources/sass/app.scss', 'public/css')
	.webpackConfig({
		plugins: [
			// reduce bundle size by ignoring moment js local files
			new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
			//ignores chinese language for the calendar
			new webpack.NormalModuleReplacementPlugin(/element-ui[\/\\]lib[\/\\]locale[\/\\]lang[\/\\]zh-CN/, 'element-ui/lib/locale/lang/en'),
			//critical css
			new Critters({
				preload: 'swap',
				preloadFonts: true
			})
		],
		//https://github.com/JeffreyWay/laravel-mix/issues/1249#issuecomment-335112415
		output: {
			publicPath: '/',
			chunkFilename: 'js/[name].[chunkhash].js',
		},
	})
	.autoExtract(); //extract all vendors
```

Thank you!

**Comment by developit** [comment]
Hello,

It appears you're using Webpack 3 or earlier, which is not supported by this plugin.

**Comment by pmochine** [comment]
@developit Thanks Jason! 

And yes I'm using Webpack 3 (cannot upgrade yet, have to wait for 5).

**Comment by developit** [comment]
Fixed and released as [2.2.0](https://github.com/GoogleChromeLabs/critters/releases/tag/2.2.0)!
